### DISCUSSION TITLE
Best directory on linux based systems to place .cab files for offline update.

### DISCUSSION THREAD

**Discussion by Aravindh09** [fwupd/fwupd#3545] [Q&A]
hi @hughsie / @superm1 ,

**Question 1:**
I have doubt on which directory to place the cab file to for an offline update. When i invoke fwupdmgr install --offline <path_to_cab_file/bios_firmware_file.cab> does fwupgmgr stages the bios_firmware_file.cab somewhere known to fwupdmgr so that in the next restart it picks bios_firmware_file.cab for update or it searches in the <path_to_cab_file/bios_firmware_file.cab> provided while invoking fwupdmgr?

if fwupmgr looks for the file in the path provided while invoking fwupdmgr then which is the best suited directory on linux based systems to place the .cab file to for the update to happen during reboot?

**Question 2:**
By invoking fwupdmgr with "install --offline </path/to/cab_file.cab>" parameter does the bios spi flash programming happens and new version of bios comes into effect in next reboot or the bios spi flash programming itself happens during the next boot only?

Thanks and Regards,
Aravindh.

**Comment by hughsie** [comment]
> does fwupgmgr stages the bios_firmware_file.cab somewhere

It should be `/var/lib/fwupd`

> does the bios spi flash programming happens

You don't need to do --offline for UEFI updates as the capsule is moved onto the EFI system partition and the BootNext is updated ready for the reboot. The only thing that needs --offline are things like superio updates that need to run in the special boot mode that's technically OS runtime, but without things like keyboard access.
